db/model/aspect.js

```diff
@@ -158,13 +158,28 @@ module.exports = function aspect(seq, dataTypes) {
 
       /**
        * Check Generator references and set the isDeleted timestamp.
+       * If aspect is deleted then send realtime "del" event to delete all
+       * samples associated with that aspect for perspectives.
        *
        * @param {Aspect} inst - The instance being destroyed
        * @returns {Promise}
        */
       beforeDestroy(inst /* , opts */) {
-        return inst.checkGeneratorReferences('delete')
-        .then(() => common.setIsDeleted(seq.Promise, inst));
+        const promiseArr = [];
+
+        promiseArr.push(redisOps.getSamplesFromAspectName(inst.name)
```

Comment:
After this operation, if we're pushing arrays onto promiseArr, isn't promiseArr an array of arrays rather than an array of promises? How does seq.Promise.all(arr) deal with an array of arrays?

Reply:
I didnt get you. We are pushing promise only.

Reply:
redisOps.getSamplesFromAspectName(inst.name) this is promise.

Reply:
oh yeah sorry i was misreading the `.each`